ovos_bus_client/util/scheduler.py

```diff
@@ -338,14 +338,21 @@ def shutdown(self):
             self.bus.remove_all_listeners('mycroft.scheduler.remove_event')
             self.bus.remove_all_listeners('mycroft.scheduler.update_event')
             # Wait for thread to finish
-            self.join()
+            self.join(30)
             # Prune event list in preparation for saving
             self.clear_repeating()
             self.clear_empty()
             # Store all pending scheduled events
             self.store()
-        except Exception as e:
-            LOG.exception(e)
+        except TimeoutError:
+            # self.join timeout
+            LOG.error("Timed out joining thread")
+        except RuntimeError as e:
+            # self.join failed
+            LOG.error(e)
+        except OSError as e:
+            # self.store failed
+            LOG.error(e)
```

Comment:
Doesn't this still leave a hung bus running? If there's nothing to be done about that, alright, but I'm not sure whether it's better to be hung or to crash.

Reply:
The bus was never closed, this only stops the EventScheduler Thread and cleans up the instance. In any case, `_stopping` was set, so there is no more looped check for event expiration so I think it makes sense to clear the `_running` signal.

This can probably be done better though